[blog] Whats new since MUI X v8 [DX-51] (#47140)
mui-org/material-ui: docs/pages/blog/mui-x-end-v8.md

---
title: MUI X v8 — Highlights since last major release

> description: Grid x Charts integration, Chart interactions, server-side Data Grid pivoting, Range Picker ergonomics, Tree View polish, and more.

Reviewer: ```suggestion
description: Grid x Charts integration, Chart interactions, server-side Data Grid pivoting, Range Picker ergonomics, Tree View improvements, and more.
```

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/ba212d3d-b5df-48a4-827d-6a42fd82f493" autoplay muted loop playsinline style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">Natural zoom and pan on desktop and touch.</figcaption>
</figure>

[Zoom and pan guide](/x/react-charts/zoom-and-pan/)

### Zoom slider with preview [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

A compact slider under the axis makes long ranges effortless to navigate, and the optional preview shows exactly what's in view—the dashboard equivalent of a map's mini-overview:

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/180cb1d1-4537-4c8f-8c7b-eb1dbba1d75e" autoplay muted loop playsinline  alt="Zoom slider and preview" style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">Stay oriented while zoomed in.</figcaption>
</figure>

[Zoom slider](/x/react-charts/zoom-and-pan/#zoom-slider)

### Keyboard navigation and a11y improvements

> Charts are more comfortable to explore from the keyboard. Focus highlights stay in sync as you move and tooltips and axes cooperate so you can read values without a mouse:

Reviewer: ```suggestion
Charts are now more comfortable to explore from the keyboard. Focus highlights stay in sync as you move, and tooltips and axes cooperate so you can read values without a mouse:
```

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/b7564569-19df-40c7-8053-91107f52f4a2" autoplay muted loop playsinline alt="Keyboard navigation in charts" style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">Keyboard-first exploration of series and points.</figcaption>
</figure>

> [Accessibility and keyboard navigation](/x/react-charts/accessibility/)

Reviewer 2: I updated the sidebar here https://github.com/mui/mui-x/pull/20700

  <figcaption style="text-align:center;margin-top:8px;">Multi-level category labels.</figcaption>
</figure>

[Axes and labeling](/x/react-charts/axis/)

### Performance and export [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

Heavy dashboards feel lighter thanks to faster string measurement, tuned bar rendering, and an SVG batch renderer for large scatter plots. When it's time to share, capture the exact view with **image export** or **print/PDF**.

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/32aec824-0c17-4350-adb5-bd72cf434c77" autoplay muted loop playsinline alt="Charts performance and export" style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">Faster charts; one-click export.</figcaption>
</figure>

[Performance tips](/x/react-charts/scatter/#performance) • [Export guide](/x/react-charts/export/)

## Data Grid

### Server-side pivoting [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

> Pivot at scale without locking the browser. The Grid coordinates with your back-end to compute pivots and aggregates, so the UI stays snappy while you slice by any model you can dream up:

Reviewer: ```suggestion
Pivot at scale without locking the browser. The Grid coordinates with your back end to compute pivots and aggregates, so the UI stays snappy while you slice by any model you can dream up:
```

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/7b6d7de3-70e9-4754-93c8-d7963d4b464a" autoplay muted loop playsinline controls style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">A clear target as you drag rows.</figcaption>
</figure>

[Row reordering docs](/x/react-data-grid/row-ordering/)

### Pinned areas and scrolling polish

Pinned rows and aggregates cooperate with the scrollbar and scroll shadows making wide, tall tables easier to parse at a glance:

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/6aa59911-d247-48c5-b576-e3fb15092e7a" autoplay muted loop playsinline alt="Pinned rows and scroll shadows" style="width:100%;max-width:960px;border-radius:12px;">
  <figcaption style="text-align:center;margin-top:8px;">Pinned context that reads at a glance.</figcaption></video>
</figure>

[Pinning guide](/x/react-data-grid/column-pinning/) • [Scrolling guide](/x/react-data-grid/scrolling/)

### Export resilience [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

> Excel export now relies on our own internally vetted and private `ExcelJS` fork to avoid transitive vulnerabilities and keep exports reliable at scale:

Reviewer: ```suggestion
Excel export now relies on our own internally vetted and private ExcelJS fork to avoid transitive vulnerabilities and keep exports reliable at scale:
```

<figure style="margin:16px 0;">
  <img src="https://github.com/user-attachments/assets/81ad75a6-5769-4a74-88f3-da45e2d2c598" alt="Export to Excel from the Data Grid" style="width:100%;max-width:960px;border-radius:12px;">
  <figcaption style="text-align:center;margin-top:8px;">Safer, sturdier exports for compliance workflows.</figcaption>
</figure>

[Export docs](/x/react-data-grid/export/)

## Date and Time Pickers

### Better range defaults [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

> Range components can use different reference dates for start and for end, so pickers open where users expect—for example, next Monday through Friday for travel, or "this month" for reporting.

Reviewer: ```suggestion
Range components can use different reference dates for starting and ending values, so pickers open where users expect—for example, next Monday through Friday for travel, or "this month" for reporting.
```

<figure style="margin:16px 0;">
  <img src="https://github.com/user-attachments/assets/595c0f74-f3b7-498b-9b0b-255f291026b1" alt="Date and time picker input polish" style="width:100%;max-width:720px;border-radius:12px;">
  <figcaption style="text-align:center;margin-top:8px;">Consistent typing, formats, and labels.</figcaption>
</figure>

[Date Pickers quickstart](/x/react-date-pickers/)

## Tree View

### Drag-and-drop support and other improvements

Between refined editing, ordering, and selection behaviors, trees hold up better as they grow. Reorder with drag-and-drop, wire up inline renaming, and keep identifiers flexible so you can mirror your domain model precisely:

<figure style="margin:16px 0;">
  <video src="https://github.com/user-attachments/assets/a91aa903-611d-43be-a361-9f933d518f55" autoplay muted loop playsinline controls style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">Scale up without losing your place.</figcaption>
</figure>

[Rich Tree View guides](/x/react-tree-view/)

## What's next

> We're targeting a v9 stable release by the end of March. After that, we'll invest heavily in **customization** and rewrite our components over Base UI to fully embrace composition, cleaner slots API, and a more modern **DevEx**. The aim is to make it easier to swap parts without forking, line up with your design tokens, and build complex UIs with less prop-surface ceremony.

Reviewer: ```suggestion
We're targeting a v9 stable release by the end of March. After that, we'll invest heavily in customization and rewrite our components on top of [Base UI](https://www.base-ui.com/) to fully embrace composition, a cleaner slots API, and a more modern developer experience The aim is to make it easier to swap parts without forking, line up with your design tokens, and build complex UIs with less prop-surface ceremony.
```

<figure style="margin:16px 0;">
  <img src="/static/blog/mui-x-end-v8/charts-perf-export.png" alt="Charts performance and export" style="width:100%;max-width:960px;border-radius:12px;">
  <figcaption style="text-align:center;margin-top:8px;">Faster charts; one-click export.</figcaption>
</figure>

[Performance tips](/x/react-charts/scatter/#performance) • [Export guide](/x/react-charts/export/)

## Data Grid

### Server-side pivoting [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

Pivot at scale without locking the browser. The Grid coordinates with your backend to compute pivots and aggregates, so the UI stays snappy while you slice by Region Segment Product (or any model you dream up).

<figure style="margin:16px 0;">
  <video src="/static/blog/mui-x-end-v8/grid-pivot.mp4" autoplay muted loop playsinline controls style="width:100%;max-width:960px;border-radius:12px;"></video>
  <figcaption style="text-align:center;margin-top:8px;">Large pivots, computed on the server.</figcaption>
</figure>

> [Start with pivoting now](/x/react-data-grid/pivoting/).

Reviewer: some of these links appear to be full sentences, others are incomplete sentences, and still others are just names. Some have periods and others don't. Can we standardize them with one style across the board?

---

> title: MUI X v8 — Highlights before the next major

Author: ```suggestion
title: MUI X v8 — Highlights since last major release
```
